lib/Frontend/CompilerInvocation.cpp

```diff
@@ -1545,3 +1545,57 @@ CompilerInvocation::loadFromSerializedAST(StringRef data) {
                         extendedInfo.getExtraClangImporterOptions().end());
   return info.status;
 }
+
+bool CompilerInvocation::getOrCreatePCH() {
+
+  const auto &BridgingHeader = getClangImporterOptions().BridgingHeader;
+
+  const auto &PCHOutputDir = getClangImporterOptions().PrecompiledHeaderOutputDir;
+
+  if (BridgingHeader.empty() || PCHOutputDir.empty()) {
+    return true;
+  }
+
+  CompilerInstance ReadInstance;
+  auto FailedToGetReadInstance = ReadInstance.setup(*this);
```

Comment:
Either. Use the Swift CompilerInstance you (will) already have and split off a custom clang::CompilerInstance within the importer. (Ideally you don't even need the separate one for reading.)